### DISCUSSION TITLE
How to read a video file with opencv

### DISCUSSION THREAD

**Discussion by AlejandroDiazD** [AxisCommunications/acap-computer-vision-sdk-examples#180] [Q&A]
Hello @Corallo ,

I'm still trying to solve the integration with Yolov5. meanwhile, I'm trying some tests with video files, but I'm not being able to open a video from a python script with opencv. I can open an image, but I'm trying to open a video following the standard opencv steps:

> import cv2
> video_path = "video_path.mp4"
> video = cv2.VideoCapture(video_path)
> video.read()

I've tried with bot codec formats, H264 and MJPEG, but always obtaining the same results:
> (False, None)

Is there any available coded on the sdk or any other way to open a video file?

Thank you!

**Comment by AlejandroDiazD** [comment]
Extending the previous question, I would have the same doubt about if is there any available codec for saving a video file from streaming using a python script, as I only saw in the object-detector-python example that images are saved but not any video. 

Thank you again!

**Comment by Corallo** [comment]
Hi @AlejandroDiazD 

We don't have an example of how to parse a video with opencv. 
It is reasonable to think that as you said it is a problem of some codec missing.
Your application container is just an Ubuntu image, if you need more codec, you can add an `RUN apt install ...` in the docker image to install what you need.

**Reply by AlejandroDiazD** [reply]
Ok, I will try that. Thank you! 

Btw, I'm also trying to use the camera streaming as input and again with wrong results...

I followed the steps of the object-detector-python example for reading input streaming video from the camera but the image I'm saving is so blurred to be used (and also seems to be horizontally duplicated 4 times in the case of setting 4:3 format from the camera interface, and 16 times in the case of 16:9).

This an example of an image received, setting 4:3 format with 1280x960 resolution:
![train_mod_results_0](https://github.com/AxisCommunications/acap-computer-vision-sdk-examples/assets/61188820/ed58668f-23c5-4b45-b215-1934d67f3fb1)

If I see the image on the web interface is perfect:
![imagen](https://github.com/AxisCommunications/acap-computer-vision-sdk-examples/assets/61188820/ad4ecd1a-629f-4fcf-8f27-76117d64fb53)


I have also tried reading the image streaming with the steps:
> import cv2
> cap = cv2.VideoCapture(0)
> cap.read()
Explained in https://axiscommunications.github.io/acap-documentation/docs/api/computer-vision-sdk-apis.html#video-capture-api

But again the results are:
> (False, None)
 
Thank you for your help!

**Reply by Corallo** [reply]
The second methods is deprecated (using opencv), the documentation update had a delay, today or tomorrow is going to be updated.
But the other thing you showed seems to be an issue.
Could you open an issue on the repository with a minimal reproducible code example? Also, please mention the resolution, camera model and firmware you are using. We'll look into it

**Reply by AlejandroDiazD** [reply]
Done! https://github.com/AxisCommunications/acap-computer-vision-sdk-examples/issues/181

Thanks!
